tests/foreman/cli/test_host_collection.py

```diff
@@ -29,257 +30,256 @@
 from robottelo.constants import ENVIRONMENT
 from robottelo.datafactory import invalid_values_list
 from robottelo.datafactory import valid_data_list
-from robottelo.test import CLITestCase
-
 
-class HostCollectionTestCase(CLITestCase):
-    """Tests for Host Collections via Hammer CLI"""
 
-    @classmethod
-    def setUpClass(cls):
-        """Prepare some data to be used in tests"""
-        super().setUpClass()
-        cls.organization = make_org()
-        cls.library = LifecycleEnvironment.info(
-            {'organization-id': cls.organization['id'], 'name': ENVIRONMENT}
-        )
-        cls.default_cv = ContentView.info(
-            {'organization-id': cls.organization['id'], 'name': DEFAULT_CV}
+def set_up_a_host_collection(module_org):
+    ''' Make a host collection for use in tests'''
+    make_host_collection({'organization-id': module_org.id})
+
+
+def _make_fake_host_helper(module_org):
+    """Make a new fake host"""
+    library = LifecycleEnvironment.info(
+        {'organization-id': module_org.id, 'name': ENVIRONMENT}
+    )
+    default_cv = ContentView.info(
+        {'organization-id': module_org.id, 'name': DEFAULT_CV}
+    )
+    return make_fake_host(
+        {
+            'content-view-id': default_cv['id'],
+            'lifecycle-environment-id': library['id'],
+            'name': gen_string('alpha', 15),
+            'organization-id': module_org.id,
+        }
+    )
+
+
+@pytest.mark.build_sanity
```

Comment:
What's the runtime of this testcase?

Reply:
robottelo]$ pytest  --show-capture=no tests/foreman/cli/test_host_collection.py -m build_sanity
=============================== test session starts ========================================
platform linux -- Python 3.8.6, pytest-5.4.3, py-1.9.0, pluggy-0.13.1
shared_function enabled - OFF - scope:  - storage: file
rootdir: /sat6/robottelo, inifile: pytest.ini
plugins: ibutsu-1.12, mock-3.3.1, forked-1.3.0, services-2.2.1, cov-2.10.1, xdist-1.34.0
collecting ... 2020-11-18 19:58:01 - conftest - DEBUG - Collected 17 test cases
collected 17 items / 16 deselected / 1 selected                                                                  

tests/foreman/cli/test_host_collection.py .                                                                [100%]

===== 1 passed, 16 deselected, 2 warnings in 156.95s (0:02:36) ======

Reply:
A little slow for what we're looking for, though on the edge of acceptable.

Would `test_positive_create_with_limit` be a sufficient option, to at least validate we can create HostCollections? If we can't update or delete a host collection, would that block us from testing the build, or require a re-build of a snap?

Reply:
> A little slow for what we're looking for, though on the edge of acceptable.
> 
> Would `test_positive_create_with_limit` be a sufficient option, to at least validate we can create HostCollections? 
Yes, will move pytest mark to use that

>If we can't update or delete a host collection, would that block us from testing the build, or require a re-build of a snap?
I guess not